### PR TITLE
Vet clinic database: query and update animals table 🥇 

### PR DESCRIPTION
**Pull request by ismailmunyentwari9**
# What  I have done in this part of project 🎄 

- [ ] Add a column `species` of type string to  animals table
- [ ] Insert new datas.
- [ ] Inside a transaction update the animals table by setting the species column to unspecified.
- [ ] Inside a transaction:
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(72).png">Update the animals table by setting the species column to pokemon for all animals that don't have species already set.</a>
- Commit the transaction.
- [ ]  <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(74).png">Inside a transaction delete all records in the animals table, then roll back the transaction.</a>
 [ ] Inside a transaction:
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(75).png">Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.</a>
- Commit transaction
- [ ] Write queries to answer the following questions:
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(76).png"> How many animals are there?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(77).png"> How many animals have never tried to escape?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(79).png">What is the average weight of animals?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(80).png">Who escapes the most, neutered or not neutered animals?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(81).png">What is the minimum and maximum weight of each type of animal?</a>
- <a href="https://github.com/ismailmunyentwari9/Vet-clinic-database/blob/animals_agrigates/Screenshot%20(82).png">What is the average number of escape attempts per animal type of those born between 1990 and 2000?</a>
# 🧠  Skills Gained in this part of the project 🌹 

- [ ] Use database transactions.
- [ ] Modify and delete data in SQL.
- [ ] Prepare complex queries that answer analytical questions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -30,5 +30,78 @@ WHERE name != 'Gabumon';
...
+/*Update the animals table by setting the species column to "digimon" for all animals that have a name ending in "mon":*/
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+/*UUpdate the animals table by setting the species column to "pokemon" for all animals that don't have species already set::*/
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
```
Comment (AnselemOdims) [inline queries.sql:59]: - [ ] Good job working on this query but please kindly include a query to verify that the change was made before committing and another the verify that the change persists after committing.

File: queries.sql [hunk 2]
```diff
@@ -30,5 +30,78 @@ WHERE name != 'Gabumon';
...
+SELECT neutered, SUM(escape_attempts) AS total_escapes
+FROM animals
+GROUP BY neutered;
+
+/* showing max and min numbers of weights between two species of animals*/
+SELECT species, max(weight_kg), min(weight_kg) from animals GROUP BY species;
+
+
+/* showing average of escape_attempts of type of animals which born  between 1900 and 200*/
+SELECT species, AVG(escape_attempts)  from animals WHERE date_of_birth BETWEEN '1900-01-01' AND '2000-01-01;'
```
Comment (AnselemOdims) [inline queries.sql]: - [ ] Good job working on this query, but please make sure it is between `1990` and `2000`. At the moment, your query is within `1900` .
Hint:
`BETWEEN '1990-01-01' AND '2000-12-31;`

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @ismailmunyentwari9 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Zuhaib042 — APPROVED** [review]
Hi @ismailmunyentwari9 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Implemented suggested changes
## Optional suggestions
- [ ] One of the key general requirements of this project is to follow Gitflow and not GitHub flow. You have created a pull request from the feature branch to the main branch which is Github flow. Kindly consider following Gitflow for all future milestones of this project.
- [ ]  There was a requirement to add screenshots of queries to the pull request description. although you added those as part of project but you should have added them in the pull request description per requirements.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**